src/excmds.ts

```diff
@@ -3373,6 +3374,76 @@ export async function ttscontrol(action: string) {
 
 //}}}
 
+// {{{ PERFORMANCE LOGGING
+
+/**
+ * Build a set of FilterConfigs from a list of human-input filter
+ * specs.
+ *
+ * @hidden
+ */
+//#background_helper
+export function buildFilterConfigs(filters: string[]): Perf.StatsFilterConfig[] {
+    return filters.map(
+        (filter: string): Perf.StatsFilterConfig => {
+            if (filter.endsWith("/")) {
+                return { kind: "ownerName", ownerName: filter.slice(0, -1) }
+            } else if (filter === ":start") {
+                return { kind: "eventType", eventType: "start" }
+            } else if (filter === ":end") {
+                return { kind: "eventType", eventType: "end" }
+            } else if (filter === ":measure") {
+                return { kind: "eventType", eventType: "measure" }
+            } else {
+                return { kind: "functionName", functionName: name }
+            }
+        },
+    )
+}
+
+/**
+ * Dump the raw json for our performance counters. Filters with
+ * trailing slashes are class names, :start | :end | :measure specify
+ * what type of sample to pass through, and all others are function
+ * names. All filters must match for a sample to be dumped.
+ *
+ * Tridactyl does not collect performance information by default. To
+ * get this data you'll have to set the configuration option
+ * `perfcounters` to `"true"`. You may also want to examine the value
+ * of `perfsamples`.
```

Comment:
This should actually activate without a reload! I renamed a function to make it slightly more clear what's going on - what used to be called `performanceEnabled` should actually be called `performanceApiAvailable`. The `config.get("perfcounters")` is actually done inside the advice, so it's tested every time the functions runs. I _think_ that `config.get` is fast enough that this should be okay, yes?

Reply:
Moved that last question to #1043

Reply:
Sorry for not answering your comments, I somehow managed to miss the notification when you wrote them.

From what I understand, decorators are called on script load and return a function which replaces the one they decorate. So if I'm right, the [`performanceApiAvailable`](https://github.com/tridactyl/tridactyl/blob/3ec27fd40c53988dd429633ca5311216d8d7e34a/src/perf.ts#L71) call is only performed on script load and if it returns false, the function is returned as-is, without performance logging. This is why I thought you needed to reload the tab/tridactyl.

I think this is what happens because the following piece of typescript
```ts
function my_decorator(a: any, b: string, c: PropertyDescriptor): PropertyDescriptor {
        console.log("decorating", a, b, c)
        return c
}

class Hello {
        @my_decorator
        world(s: string) {
                return s
        }
}
```
Is compiled into the following piece of JavaScript:
```js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") {
        r = Reflect.decorate(decorators, target, key, desc);
    } else {
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    }
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function my_decorator(a, b, c) {
    console.log("decorating", a, b, c);
    return c;
}
var Hello = /** @class */ (function () {
    function Hello() {
    }
    Hello.prototype.world = function (s) {
        return s;
    };
    __decorate([
        my_decorator
    ], Hello.prototype, "world");
    return Hello;
}());
```
As you can see, `__decorate` is called once when creating Hello. It calls `my_decorator` immediately and then assigns its return value to Hello's prototype.

Am I missing/misunderstanding something?

Reply:
Disregard everything I've said - I didn't understand the code well enough and some of my assumptions were wrong.

Reply:
Yep, thanks for taking the time to clarify that :)